sdk/lib/drivers/sound/mmebuddy/wave/streaming.c

```diff
@@ -206,6 +212,15 @@ CompleteIO(
             /* Partially completed */
             HdrExtension->BytesCompleted += dwNumberOfBytesTransferred;
             SND_TRACE(L"%d/%d bytes of wavehdr completed\n", HdrExtension->BytesCompleted, WaveHdr->dwBufferLength);
+
+            /* Do we loop a header? */
+            if (HdrExtension->BytesCommitted == WaveHdr->dwBufferLength &&
+                SoundOverlapped->PerformCompletion == FALSE)
+            {
+                /* Reset amount of bytes and decrement loop count, to play next iteration */
+                HdrExtension->BytesCommitted = 0;
+                --SoundDeviceInstance->LoopsRemaining;
```

Comment:
Are we sure here, that LoopsRemaining is > 0 ? (I guess, if the == 0 case is when PerformCompletion is TRUE ; I would still like to see an `ASSERT(SoundDeviceInstance->LoopsRemaining > 0);` just before the decrement.)

Reply:
We have `SND_ASSERT` internal macro in mmebuddy, which is preferred to be used here. So I'll use it instead.

Reply:
Committed.